Detectors/FIT/simulation/src/Digitizer.cxx

```diff
@@ -82,127 +73,118 @@ void Digitizer::process(const std::vector<HitType>* hits, Digit* digit)
+    cfd[mcp] = d.CFDTime; 
+    amp[mcp] = d.QTCAmpl;
   }
-  // --------------------------------------------------------------------------
-
-  //Calculating signal time, amplitude in mean_time +- time_gate --------------
-  Double_t ch_signal_nPe[nMCPs] = {};
-  Double_t ch_signal_MIP[nMCPs] = {};
-  Double_t ch_signal_time[nMCPs] = {};
 
-  for (auto& hit : *hits) {
-    Int_t hit_ch = hit.GetDetectorID();
-    Double_t hit_time = hit.GetTime();
-    Bool_t is_A_side = (hit_ch <= 4 * Geometry::NCellsA);
-    Double_t time_compensate = is_A_side ? A_side_cable_cmps : C_side_cable_cmps;
-    Bool_t is_hit_in_signal_gate = (hit_time > ch_hit_mean_time[hit_ch] - signal_width * .5) &&
-                                   (hit_time < ch_hit_mean_time[hit_ch] + signal_width * .5);
-
-    Double_t hit_time_corr = hit_time - time_compensate + BC_clk_center /* + BCEventTime*/;
-    Double_t is_time_in_gate = (hit_time != 0.); //&&(hit_time_corr > -BC_clk_center)&&(hit_time_corr < BC_clk_center);
-
-    if (is_time_in_gate && is_hit_in_signal_gate) {
-      ch_signal_nPe[hit_ch]++;
-      ch_signal_time[hit_ch] += hit_time_corr;
+  for (Int_t ch_iter = 0; ch_iter < mMCPs; ch_iter++) {
+    if (ch_signal_nPe[ch_iter] != 0) {
+      ch_signal_MIP[ch_iter] = amp[ch_iter] + ch_signal_nPe[ch_iter] / nPe_in_mip;
+      if (cfd[ch_iter] > 0) {
+        cfd[ch_iter] = cfd[ch_iter] - mBC_clk_center - mEventTime;
+        ch_signal_time[ch_iter] = ((cfd[ch_iter] + ch_signal_time[ch_iter] / (float)ch_signal_nPe[ch_iter]) / 2.) + mBC_clk_center + mEventTime;
+      } else
+        ch_signal_time[ch_iter] = (ch_signal_time[ch_iter] / (float)ch_signal_nPe[ch_iter]) + mBC_clk_center + mEventTime;
+
+      if (ch_signal_MIP[ch_iter] > mCFD_trsh_mip) {
+        LOG(DEBUG) << ch_iter << " : "
+                   << " : " << ch_signal_time[ch_iter] - mBC_clk_center - mEventTime << " : "
+                   << ch_signal_MIP[ch_iter] << " : " << mEventTime << " cfd " << cfd[ch_iter] << FairLogger::endl;
+        mChDgDataArr.emplace_back(ChannelData{ ch_iter, ch_signal_time[ch_iter], ch_signal_MIP[ch_iter] });
+      } else {
+        ch_signal_MIP[ch_iter] = 0;
+        ch_signal_time[ch_iter] = 0;
+      }
     }
   }
 
-  for (Int_t ch_iter = 0; ch_iter < nMCPs; ch_iter++) {
-    if (ch_signal_nPe[ch_iter] != 0) {
-      ch_signal_MIP[ch_iter] = ch_signal_nPe[ch_iter] / nPe_in_mip;
-      ch_signal_time[ch_iter] = ch_signal_time[ch_iter] / (float)ch_signal_nPe[ch_iter];
-    }
+  digit->setChDgData(std::move(mChDgDataArr));
+  
+}
 
-    //if ampl less than cfd trh adc and cfd has no signal
-    if (ch_signal_MIP[ch_iter] < CFD_trsh_mip) {
-      ch_signal_MIP[ch_iter] = 0.;
-      ch_signal_time[ch_iter] = 0.;
+//------------------------------------------------------------------------
+void Digitizer::smearCFDtime(Digit* digit)
+{
+  //smeared CFD time for 50ps
+  std::vector<ChannelData> mChDgDataArr;
+  for (const auto& d : digit->getChDgData()) {
```

Comment:
I don't understand this code: you smear only the stack mChDgDataArr vector, w/o any effect on the digit or the digitizer state